### PR TITLE
Pla 13613/add catalyst insights support

### PR DESCRIPTION
**Pull request by StevenKauwe**
# Citrine Python PR

## Description 
Please briefly explain the goal of the changes/this PR.
The reviewer should be able to understand why the change is being made by reading this description
and its links (e.g. JIRA tickets).

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [x] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in __version\__.py

### CODE REVIEW

**Comment by StevenKauwe** [comment]
@anoto-moniz, sorry for the reformatting. I have black enabled and didn't realize till too late 😆  I didn't change your code, except using the language model enums.

**Comment by anoto-moniz** [comment]
> @anoto-moniz, sorry for the reformatting. I have black enabled and didn't realize till too late 😆 I didn't change your code, except using the language model enums.

Could you please reset it? Because although we don't have our linter set to super strictly check for it, `black` ends up with a style that's annoyingly disparate from the rest of the codebase. The indenting and such (which I'm actually surprised `flake8` didn't yell about...)

File: src/citrine/informatics/catalyst/language_model.py
Comment (anoto-moniz) [inline src/citrine/informatics/catalyst/language_model.py:1]: Good call! I should have done this. :sweat_smile:

File: src/citrine/resources/catalyst.py [hunk 1]
```diff
@@ -35,11 +36,39 @@ def assistant(self, query: str, *, predictor: GraphPredictor) -> AssistantRespon
...
+        The insights response, containing details of the result of your query which vary by type.
+
+        """
+        user = UserCollection(self.session).me()
+        if not user.is_internal:
+            raise NotImplementedError(
+                "I'm sorry, but this feature is not currently available for "
+                "your organization. Please watch the release notes for "
+                "updates."
+            )
```
Comment (anoto-moniz) [inline src/citrine/resources/catalyst.py:69]: Let's abstract this into a helper to reduce duplication.

File: tests/resources/test_catalyst.py [hunk 2]
```diff
@@ -173,21 +169,50 @@ def assistant_predictor(assistant_predictor_data):
...
+def insights_response_data():
+    return {
+        "response": "Here is the insight based on the literature.",
+        "relevant_documents": True,
+        "document_metadata": [{"source": "doi:10.1234/example", "source_type": "doi"}],
+    }
+
+
+@pytest.fixture
+def insights_response_failed_data():
```
Comment (anoto-moniz) [inline tests/resources/test_catalyst.py:182]: You're not using this fixture anywhere. And it's malformed, as it's a dict within a dict, without a key.

File: tests/resources/test_catalyst.py [hunk 3]
```diff
@@ -173,21 +169,50 @@ def assistant_predictor(assistant_predictor_data):
...
+    return {
+        {
+            "response": "Error during insight generation.",
+            "relevant_documents": False,
+            "document_metadata": [],
+        }
+    }
```
Comment (anoto-moniz) [inline tests/resources/test_catalyst.py:189]: You're not using this fixture anywhere. And it's malformed, as it's a dict within a dict, without a key.

File: src/citrine/informatics/catalyst/insights.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    )
+
+    def __init__(
+        self,
+        *,
+        question: str,
+        temperature: Optional[float] = 0.0,
+        language_model: Optional[
+            LanguageModelChoice
+        ] = LanguageModelChoice.GPT_35_TURBO,
```
Comment (anoto-moniz) [inline src/citrine/informatics/catalyst/insights.py:30]: This formatting is super gross and confusing. It's stuff like this why I'm skeptical of tools like `black`...

**Review by anoto-moniz — APPROVED** [review]
Thanks!

**Pull request merged**